src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp

```diff
@@ -36,14 +40,21 @@ QgsExternalResourceConfigDlg::QgsExternalResourceConfigDlg( QgsVectorLayer *vl,
 
   mRootPath->setPlaceholderText( QgsSettings().value( QStringLiteral( "/UI/lastExternalResourceWidgetDefaultPath" ), QDir::toNativeSeparators( QDir::cleanPath( defpath ) ) ).toString() );
 
-  // Add connection to button for choosing default path
-  connect( mRootPathButton, SIGNAL( clicked() ), this, SLOT( chooseDefaultPath() ) );
+  QMenu *rootPathMenu = new QMenu();
+  mRootPathExpressionButton = new QAction( QgsApplication::getThemeIcon( QStringLiteral( "/mIconExpression.svg" ) ), tr( "Expression" ), rootPathMenu );
```

Comment:
icon doesn't seem to show up properly
![image](https://cloud.githubusercontent.com/assets/127259/23607950/3f15fe14-0267-11e7-9347-910156fc9d50.png)

do you confirm?